### PR TITLE
Riley/c 2822 inbox doesnt refetch unreadmessagecount

### PR DESCRIPTION
**Pull request by rileylnapier**
## Description

make sure to refetch when we get new userId or clientKey and make sure to update graphqlclient
## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> `Fix [#1]()`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,15 +2,22 @@
 [![stability-beta](https://img.shields.io/badge/stability-beta-33bbff.svg?label=Stability)](https://github.com/mkenney/software-guides/blob/master/STABILITY-BADGES.md#beta)
 [![codecov](https://codecov.io/gh/trycourier/courier-react/branch/main/graph/badge.svg?token=NVTDWY9UH4)](https://codecov.io/gh/trycourier/courier-react)
 
-# Courier React Components (CRC)
+## What is `courier-react`?
 
-CRC is a monorepo using [Lerna](https://www.lerna.com) and contains many packages that are installed separately while depending on each other.
+This repository, also called **In-App**, is a set of features and components that can work inside our customers websites. Courier is a powerful application but integrating all of the features into your frontend manually can be overwhelming and time consuming. This is where In-App steps up and can superpower your application by providing Realtime Messaging (Toast), an Inbox, Notification Preferences, and many more coming soon!
```
Comment (suhasdeshpande) [inline README.md:7]:
```suggestion
This repository, also called **In-App**, is a set of features and components that can work inside our customers web applications. Courier is a powerful application but integrating all of the features into your frontend manually can be overwhelming and time consuming. This is where In-App steps up and can superpower your application by providing Realtime Messaging (Toast), an Inbox, Notification Preferences, and many more coming soon!
```

File: packages/react-provider/src/reducer.ts [hunk 2]
```diff
@@ -29,11 +29,12 @@ const rootReducer = (state, action) => {
       return {
         ...state,
         apiUrl: action.payload.apiUrl,
+        brandId: action.payload.brandId,
         clientKey: action.payload.clientKey,
+        graphQLClient: action.payload.graphQLClient,
```
Comment (suhasdeshpande) [inline packages/react-provider/src/reducer.ts:34]: curious why we need graphQLClient here if its available on context?

**Review by suhasdeshpande — APPROVED** [review]
SFTM, one question about reducer and graphql client

File: packages/react-provider/src/reducer.ts (hunk 2 above)
Comment (rileylnapier) [reply packages/react-provider/src/reducer.ts:34]: because it can changed based on the clientKey and userId

**Pull request merged**